webapp/src/components/badge.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import {OverlayTrigger} from 'react-bootstrap';
+import styled, {css} from 'styled-components';
+
+export type Props = {
+    text: string,
+    textSize: number,
+    icon: React.ReactNode,
+    gap: number,
+    iconFill?: string,
+    bgColor?: string,
+    margin?: string,
+    padding?: string,
+    loading?: boolean,
+    color?: string,
+}
+
+export default function Badge(props: Props) {
+    return (
+        <Container
+            bgColor={props.bgColor}
+            size={props.textSize}
+            margin={props.margin}
+            padding={props.padding}
+            color={props.color}
+        >
+            { props.loading &&
+            <Spinner size={props.textSize}/>
+            }
+            { !props.loading &&
+            <Icon fill={props.iconFill}>{props.icon}</Icon>
+            }
+            <Text gap={props.gap}>{props.text}</Text>
+        </Container>
+    );
+}
+
+const Container = styled.div<{bgColor?: string, size: number, margin?: string, padding?: string, color?: string}>`
```

Comment:
This would be cleaner if we had a `ContainerProps` interface, imo.

Reply:
Yeah, it was originally one thing, then it became two and here we are. I'll fix it :+1: